ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/StringTemplateRule.kt

```diff
@@ -65,9 +69,19 @@ class StringTemplateRule : Rule("string-template") {
             ) {
                 emit(node.treePrev.startOffset + 2, "Redundant curly braces", true)
                 if (autoCorrect) {
-                    // fixme: a proper way would be to downcast to SHORT_STRING_TEMPLATE_ENTRY
-                    (node.firstChildNode as LeafPsiElement).rawReplaceWithText("$") // entry start
-                    (node.lastChildNode as LeafPsiElement).rawReplaceWithText("") // entry end
+                    val leftCurlyBraceNode = node.findChildByType(LONG_TEMPLATE_ENTRY_START)
+                    val rightCurlyBraceNode = node.findChildByType(LONG_TEMPLATE_ENTRY_END)
+                    if (node.children().count() == 3 && leftCurlyBraceNode != null && rightCurlyBraceNode != null) {
```

Comment:
Could you elaborate on why `count` is 3 exactly? I guess I am missing a bit of context :)

Reply:
It is a bit of defensive programma. The block above is surrounded by the following if-statement:
```
            if (node.text.startsWith("${'$'}{") &&
                node.text.let { it.substring(2, it.length - 1) }.all { it.isPartOfIdentifier() } &&
                (entryExpression is KtNameReferenceExpression || entryExpression is KtThisExpression) &&
                node.treeNext.let { nextSibling ->
                    nextSibling.elementType == CLOSING_QUOTE ||
                        (
                            nextSibling.elementType == LITERAL_STRING_TEMPLATE_ENTRY &&
                                !nextSibling.text[0].isPartOfIdentifier()
                            )
                }
            ) {
```
I was not 100% sure that this statement really covers the situation of a prefix-node, content-node and a suffix-node. In case that also prefix-node, content-node-1, ...,  content-node-n, suffix-node can match this statement then I do think it is not a valid to prefix only the first node with the '$'. So line 74 guards the autocorrect to patterns which can be corrected safely.